planner/core/preprocess.go

```diff
@@ -1176,8 +1181,8 @@ func checkDuplicateColumnName(IndexPartSpecifications []*ast.IndexPartSpecificat
 }
 
 // checkIndexInfo checks index name, index column names and prefix lengths.
-func checkIndexInfo(indexName string, IndexPartSpecifications []*ast.IndexPartSpecification) error {
-	if strings.EqualFold(indexName, mysql.PrimaryKeyName) {
+func checkIndexInfo(indexName string, IndexPartSpecifications []*ast.IndexPartSpecification, isPK bool) error {
+	if strings.EqualFold(indexName, mysql.PrimaryKeyName) && !isPK {
```

Comment:
```suggestion
	if !isPK && strings.EqualFold(indexName, mysql.PrimaryKeyName) {
```
(really minor, but usually better to do cheap comparisons first)